### ISSUE TITLE
Window not defined in SSR

### ISSUE THREAD

**Issue by stephenbaldwin** [rcaferati/react-awesome-slider#26]
Importing react-awesome-slider on server throws a window not defined error

**Comment by Zeroxys** [comment]
any answer about how solve this ?

**Comment by csprance** [comment]
If you are using something like nextjs you can do

```ts
import dynamic from 'next/dynamic';
import * as React from 'react';

/*
This requires access to the window so we need to dynamically import it with no SSR
 */
const AwesomeSlider = dynamic(() => import('react-awesome-slider'), {
  ssr: false
});

export default AwesomeSlider;
```

and then import the css and use it like normal.

**Comment by hollowaysmith** [comment]
I ended up copying the build file and adding some safety checks to usage of window. There were only about 10 changes needed to make it work. Copy this file to your project and import it instead of the `react-awesome-slider` NPM distribution.

https://gist.github.com/hollowaysmith/c359b0530420ac955108e8579bddd537

It works for me in Gatsby with this change.

@rcaferati SSR just needs safety checks around `window` before using it. Ie,

before:
```
window.requestAnimationFrame(() => {  (stuff) }
```
after:
```
window && window.requestAnimationFrame(() => {  (stuff) }
```

I'd prefer to use your distribution if this was fixed :smile:
